experimental/adaptive-dialog/docs/memory-model-overview.md

```diff
@@ -31,33 +39,54 @@ Examples:
+Dialog scope is persistent data scoped for a giving executing dialog, providing an memory space for each dialog to have internal persistent bookkeeping. Dialog scope is cleared when dialog ends. You can access properties available under the `dialog.XXX` scope via `$XXX`. `$` is a shorthand for `dialog` memory scope.
 
 Examples:
-
+```markdown
     dialog.orderStarted
+
+    > same as 
+    $orderStarted
+
     dialog.shoppingCart
 
+    > same as
+    $shoppingCart
+```
+
 ### Dialog sub-scopes
-Two interesting sub-scopes under dialog are 'result' and 'options'. Options scope by default carry parameters/ input to the specific dialog being executed. Result scope captures any values returned by a dialog. You can access these via dialog.result and dialog.options.
+- `dialog.options` scope by default carry parameters/ input to the specific dialog being executed. 
+- `dialog.foreach` scope by default carry dialog.foreach.value and dialog.foreach.index and are available to actions within a `forEach` action. 
 
 ## Turn scope
 Turn scope is *non-persistent* data scoped for *only the current turn*, providing a place to share data for the lifetime of the current turn.  
 
 Examples:
 
-    turn.DialogEvent
-    turn.recognitionResult
+    turn.bookingConfirmation
+    turn.actvityProcessed
 
 ### Turn sub-scopes
-- Each incoming [activity][5] to the bot is available via turn.activity scope.
-- For each turn of the conversation, if a [recognizer][4] is run, then the output intents and entities from that recognizer are automatically set and available under _turn.recognized.intents_ and _turn.recognized.entities_
-- The payload of an event raised by the system (or via user code) is available under turn.dialogEvents.\<eventName\>.value scope.
+- `turn.activity` - Each incoming [activity][5] to the bot is available via turn.activity scope.
+- `turn.recognized` For each turn of the conversation, if a [recognizer][4] is run, then the output intents and entities from that recognizer are automatically set. 
+    - `turn.recognized.intents.xxx` - path to the top intent classified by the recognizer for that turn.
+    - `turn.recognized.entities.xxx` - path to entities recognized for that turn.
+    - `turn.recognized.score` - path to the top scoring intent's confidence score.
+- `turn.dialogEvent` - The payload of an event raised by the system (or via user code) is available under turn.dialogEvent.\<eventName\>.value scope.
+- `turn.lastresult` - The result from the last dialog that was called.
+- `turn.activityProcessed` - Bool property which if set means that the turncontet.activity has been consumed by some component in the system.
 
 ## Settings scope
 This represents any settings that are made available to the bot via the platform specific settings configuration system - e.g. appsettings.json, .env etc. 
```

Comment:
".,..  - e.g. appsettings.json, .env, **dynamic environment setting in Azure** etc

Reply:
Fixed.